src/satosa/backends/saml2.py

```diff
@@ -139,6 +140,13 @@ def authn_request(self, context, entity_id):
                            exc_info=True)
             raise SATOSAAuthenticationError(context.state, "Failed to construct the AuthnRequest") from exc
 
+        if self.sp.config._sp_allow_unsolicited is False:
```

Comment:
The PR passes the tests now, but this right here (peeking at `_`-prefixed/internal data) is really ugly. It should be fixed in `pysaml2` (`config.py`). I will have a look.